python/lsst/sconsUtils/installation.py

```diff
@@ -213,28 +219,31 @@ def Declare(self, products=None):
             products = [None]
 
         for prod in products:
-            if not prod or isinstance(prod, str):   # i.e. no version
+            if not prod or isinstance(prod, str):  # i.e. no version
                 product = prod
 
-                if 'version' in self:
-                    version = self['version']
+                if "version" in self:
+                    version = self["version"]
                 else:
                     version = None
             else:
                 product, version = prod
 
             if not product:
-                product = self['eupsProduct']
+                product = self["eupsProduct"]
 
             if "EUPS_DIR" in os.environ:
-                self['ENV']['PATH'] += os.pathsep + "%s/bin" % (os.environ["EUPS_DIR"])
+                self["ENV"]["PATH"] += os.pathsep + "%s/bin" % (os.environ["EUPS_DIR"])
                 self["ENV"]["EUPS_LOCK_PID"] = os.environ.get("EUPS_LOCK_PID", "-1")
                 if "undeclare" in SCons.Script.COMMAND_LINE_TARGETS or self.GetOption("clean"):
                     if version:
-                        command = "eups undeclare --flavor %s %s %s" % \
-                                  (self['eupsFlavor'], product, version)
-                        if ("current" in SCons.Script.COMMAND_LINE_TARGETS
-                                and "declare" not in SCons.Script.COMMAND_LINE_TARGETS):
+                        command = "eups undeclare --flavor {} {} {}".format(
```

Comment:
Interesting that this didn't go f-string.  (You can say `f"eups undeclare --flavor {self['eupsFlavor']} ..."`, although I don't know what `black` does to it.)

Reply:
Fixed it.